sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.JsonSchema/src/SchemaRegistryJsonSerializer.cs

```diff
@@ -0,0 +1,610 @@
+        ///   The <see cref="Exception.InnerException"/> will contain the underlying exception from the <see cref="SchemaRegistryJsonSchemaGenerator"/> class.
+        /// </exception>
+        public object Deserialize(
+            MessageContent content,
+            Type dataType,
+            CancellationToken cancellationToken = default)
+            => DeserializeMessageDataInternalAsync(content.Data, dataType, content.ContentType, false, cancellationToken).EnsureCompleted();
+
+        /// <summary>
+        /// Deserializes the message data into the specified type using the schema information populated in <see cref="MessageContent.ContentType"/>.
+        /// </summary>
+        /// <param name="content">The message containing the data to deserialize.</param>
+        /// <param name="dataType">The type to deserialize the message data into.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        /// <returns>The deserialized data.</returns>
+        /// <exception cref="FormatException">
+        ///   The ContentType is not in the expected format. The ContentType is expected to be 'application/json+schema-id', where 'schema-id' is
+        ///   the Schema Registry schema ID.
+        /// </exception>
+        /// <exception cref="RequestFailedException">
+        ///   An error occurred while attempting to communicate with the Schema Registry service.
+        /// </exception>
+        /// <exception cref="Exception">
+        ///   The schema from <paramref name="dataType"/> was not compatible with the schema used to serialize the data, or the schema itself was invalid.
+        ///   The <see cref="Exception.InnerException"/> will contain the underlying exception from the <see cref="SchemaRegistryJsonSchemaGenerator"/> class.
+        /// </exception>
+        public async ValueTask<object> DeserializeAsync(
+            MessageContent content,
+            Type dataType,
+            CancellationToken cancellationToken = default)
+            => await DeserializeMessageDataInternalAsync(content.Data, dataType, content.ContentType, true, cancellationToken).ConfigureAwait(false);
+
+        private async ValueTask<object> DeserializeMessageDataInternalAsync(
+            BinaryData data,
+            Type dataType,
+            ContentType? contentType,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            Argument.AssertNotNull(data, nameof(data));
+            Argument.AssertNotNull(contentType, nameof(contentType));
+
+            string[] contentTypeArray = contentType.ToString().Split('+');
+            if (contentTypeArray.Length != 2 || contentTypeArray[0] != JsonMimeType)
+            {
+                throw new FormatException("Content type was not in the expected format of 'application/json+schema-id', where 'schema-id' " +
```

Comment:
Since you've got constants for the mime type, I'd suggest we use that here rather than hardcoding.